tests/scene/test_timer.h

```diff
@@ -0,0 +1,150 @@
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef TEST_TIMER_H
+#define TEST_TIMER_H
+
+#include "scene/main/timer.h"
+
+#include "tests/test_macros.h"
+
+namespace TestTimer {
+
+TEST_CASE("[SceneTree][Timer] Check Timer Setters and Getters") {
+	Timer *test_timer = memnew(Timer);
+
+	SUBCASE("[Timer] Timer set and get wait time") {
+		// check default
+		CHECK(Math::is_equal_approx(test_timer->get_wait_time(), 1.0));
+
+		test_timer->set_wait_time(50.0);
+		CHECK(Math::is_equal_approx(test_timer->get_wait_time(), 50.0));
+
+		test_timer->set_wait_time(42.0);
+		CHECK(Math::is_equal_approx(test_timer->get_wait_time(), 42.0));
+
+		// wait time remains unchanged if we attempt to set it negative or zero
+		ERR_PRINT_OFF;
+		test_timer->set_wait_time(-22.0);
+		ERR_PRINT_ON;
+		CHECK(Math::is_equal_approx(test_timer->get_wait_time(), 42.0));
+
+		ERR_PRINT_OFF;
+		test_timer->set_wait_time(0.0);
+		ERR_PRINT_ON;
+		CHECK(Math::is_equal_approx(test_timer->get_wait_time(), 42.0));
+	}
+
+	SUBCASE("[Timer] Timer set and get one shot") {
+		// check default
+		CHECK(test_timer->is_one_shot() == false);
+
+		test_timer->set_one_shot(true);
+		CHECK(test_timer->is_one_shot() == true);
+
+		test_timer->set_one_shot(false);
+		CHECK(test_timer->is_one_shot() == false);
+	}
+
+	SUBCASE("[Timer] Timer set and get autostart") {
+		// check default
+		CHECK(test_timer->has_autostart() == false);
+
+		test_timer->set_autostart(true);
+		CHECK(test_timer->has_autostart() == true);
+
+		test_timer->set_autostart(false);
+		CHECK(test_timer->has_autostart() == false);
+	}
+
+	SUBCASE("[Timer] Timer start and stop") {
+		test_timer->set_autostart(false);
+	}
+
+	SUBCASE("[Timer] Timer set and get paused") {
+		// check default
+		CHECK(test_timer->is_paused() == false);
+
+		test_timer->set_paused(true);
+		CHECK(test_timer->is_paused() == true);
+
+		test_timer->set_paused(false);
+		CHECK(test_timer->is_paused() == false);
+	}
+
+	memdelete(test_timer);
+}
+
+TEST_CASE("[SceneTree][Timer] Check Timer Start and Stop") {
+	Timer *test_timer = memnew(Timer);
+
+	SUBCASE("[Timer] Timer start and stop") {
+		// @TODO: how do i add the timer to a tree?
```

Comment:
I've seen the Node tests do it like this:
```cpp
SceneTree::get_singleton()->get_root()->add_child(node);
```

Reply:
Ah, thank you.